ccmlib/scylla_node.py

```diff
@@ -514,26 +529,31 @@ def stop(self, wait=True, wait_other_notice=False, other_nodes=None, gently=True
                     except OSError:
                         pass
 
+            if not wait and not wait_other_notice:
+                return True
+
+            time.sleep(1)
+            if not self.wait_until_stopped(wait_seconds):
+                if gently and self.pid:
+                    # Aborting is intended to generate a core dump
+                    # so the reason the node didn't stop normally can be studied.
+                    print("{} is still running. Trying to kill it with SIGQUIT...".format(self))
+                    try:
+                        os.kill(self.pid, signal.SIGQUIT)
+                    except OSError:
+                        pass
+                    if self.jmx_pid:
+                        try:
+                            os.kill(self.jmx_pid, signal.SIGKILL)
+                        except OSError:
+                            pass
+                    self.wait_until_stopped(wait_seconds)
+                if self.is_running():
+                    raise NodeError("Problem stopping node %s" % self.name)
+
             if wait_other_notice:
                 for node, mark in marks:
                     node.watch_log_for_death(self, from_mark=mark)
-            else:
```

Comment:
Why switch the order ? first killing the node then waiting for others ?

Reply:
Others won't notice if the node hangs during shutdown,
so we want to force kill it if needed and then look if the others noticed.